CommandWrapper#executeAsyncWithMetrics fails with ClassCastException

**Describe the bug**

Some spring client metric recordings fail, e.g. Job Completion commands, basically all that have not the ResponseType Void, which [is statically used in CommandWrapper](https://github.com/camunda/camunda/blob/main/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/jobhandling/CommandWrapper.java#L28).

RuntimeException:
```
java.util.concurrent.CompletionException: java.lang.ClassCastException: class io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl cannot be cast to class java.lang.Void (io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl is in unnamed module of loader 'app'; java.lang.Void is in module java.base of loader 'bootstrap')
```

The issue was introduced with https://github.com/camunda/camunda/commit/08549f1a333b2d79a435953883742676163f619a#diff-72dbd98a316339b0694a986876f59994f004caaba30d0607d865752449c66c68R61
where metric handling was moved inside the CommandWrapper using `thenApply` which casts the result to "Void" at runtime, which it may not be compatible with.

Relates to #26424 where I encountered the issue it when running tests.

**To Reproduce**

Use Spring SDK and complete jobs, job completion metrics are not updated due to the exception. General client usage is not impacted as the error is swallowed in the `CommandWrapper` implementation

**Expected behavior**

MetricsRecorder should record job completion metrics.

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.concurrent.CompletionException: java.lang.ClassCastException: class io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl cannot be cast to class java.lang.Void (io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl is in unnamed module of loader 'app'; java.lang.Void is in module java.base of loader 'bootstrap')
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.onNext(ZeebeClientFutureImpl.java:85)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:468)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInternal(ClientCallImpl.java:657)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:644)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.ClassCastException: class io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl cannot be cast to class java.lang.Void (io.camunda.zeebe.client.impl.response.CompleteJobResponseImpl is in unnamed module of loader 'app'; java.lang.Void is in module java.base of loader 'bootstrap')
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	... 11 more

```

</p>
</details>

**Environment:**
- Zeebe Version: 8.6+
- Configuration: Spring SDK client

--- a/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/jobhandling/CommandWrapper.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/jobhandling/CommandWrapper.java
@@ -25,7 +25,7 @@ import java.util.concurrent.TimeUnit;
 
 public class CommandWrapper {
 
-  private final FinalCommandStep<Void> command;
+  private final FinalCommandStep<?> command;
   private final ActivatedJob job;
   private final CommandExceptionHandlingStrategy commandExceptionHandlingStrategy;
   private final MetricsRecorder metricsRecorder;
@@ -35,7 +35,7 @@ public class CommandWrapper {
   private final int maxRetries;
 
   public CommandWrapper(
-      final FinalCommandStep<Void> command,
+      final FinalCommandStep<?> command,
       final ActivatedJob job,
       final CommandExceptionHandlingStrategy commandExceptionHandlingStrategy,
       final MetricsRecorder metricsRecorder,

--- a/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/jobhandling/JobHandlerInvokingSpringBeans.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/jobhandling/JobHandlerInvokingSpringBeans.java
@@ -19,6 +19,7 @@ import io.camunda.client.api.command.CompleteJobCommandStep1;
 import io.camunda.client.api.command.FinalCommandStep;
 import io.camunda.client.api.command.ThrowErrorCommandStep1.ThrowErrorCommandStep2;
 import io.camunda.client.api.response.ActivatedJob;
+import io.camunda.client.api.response.CompleteJobResponse;
 import io.camunda.client.api.worker.JobClient;
 import io.camunda.client.api.worker.JobHandler;
 import io.camunda.client.impl.Loggers;
@@ -116,7 +117,7 @@ public class JobHandlerInvokingSpringBeans implements JobHandler {
     return parameterResolvers.stream().map(resolver -> resolver.resolve(jobClient, job)).toList();
   }
 
-  private FinalCommandStep createCompleteCommand(
+  private FinalCommandStep<CompleteJobResponse> createCompleteCommand(
       final JobClient jobClient, final ActivatedJob job, final Object result) {
     CompleteJobCommandStep1 completeCommand = jobClient.newCompleteCommand(job.getKey());
     if (result != null) {

